tests/unit_tests/test_sky/server/requests/test_requests.py

```diff
@@ -1545,3 +1541,195 @@ async def test_get_latest_request_id_async(isolated_database):
+                                            status=RequestStatus.SUCCEEDED,
+                                            created_at=current_time + 100,
+                                            user_id='other-user')
+    await requests.create_if_not_exists_async(non_matching_request)
+
+    # Test with non-matching prefix
+    result = requests.get_requests_with_prefix('nonexistent-prefix')
+    assert result is None
+
+    # Test with prefix that matches exactly one request
+    result = requests.get_requests_with_prefix('batch-request-000')
+    assert result is not None
+    assert len(result) == 1
+    assert result[0].request_id == 'batch-request-000'
+    assert result[0].name == 'test-request-0'
+    assert result[0].user_id == 'test-user-0'
+    assert result[0].cluster_name == 'cluster-0'
+    assert result[0].status == RequestStatus.PENDING
+
+    # Test with prefix that matches multiple requests
+    result = requests.get_requests_with_prefix('batch-request')
+    assert result is not None
+    assert len(result) == 3
+
+    # Verify all returned requests match the prefix
+    returned_ids = [req.request_id for req in result]
+    expected_ids = [
+        'batch-request-000', 'batch-request-001', 'batch-request-002'
+    ]
+    assert set(returned_ids) == set(expected_ids)
+
+    # Verify request details
+    for req in result:
+        assert req.request_id.startswith('batch-request')
+        assert req.name.startswith('test-request')
+        assert req.user_id.startswith('test-user')
+
+    # Test with empty prefix (should match all requests)
+    result = requests.get_requests_with_prefix('')
+    assert result is not None
+    assert len(result) == 4
+    returned_ids = [req.request_id for req in result]
+    expected_ids = [
+        'batch-request-000', 'batch-request-001', 'batch-request-002',
+        'other-request-1'
+    ]
+    assert set(returned_ids) == set(expected_ids)
+
+    # Test with specific fields - only request_id and name
+    result = requests.get_requests_with_prefix('batch-request',
+                                               fields=['request_id', 'name'])
+    assert result is not None
+    assert len(result) == 3
+
+    # Verify that only the requested fields are meaningful
+    for req in result:
+        assert req.request_id in [
+            'batch-request-000', 'batch-request-001', 'batch-request-002'
+        ]
+        if req.request_id == 'batch-request-000':
+            assert req.name == 'test-request-0'
+        elif req.request_id == 'batch-request-001':
+            assert req.name == 'test-request-1'
+        else:
+            assert req.name == 'test-request-2'
+        assert req.pid is None
+        assert req.finished_at is None
+        assert req.should_retry is False
+        assert req.status_msg is None
+
+
+@pytest.mark.asyncio
+async def test_get_requests_async_with_prefix(isolated_database):
```

Comment:
Can we merge this together with `test_get_requests_with_prefix`?

I think we should be able to use smth like `@pytest.mark.parametrize('async', [True, False])`, and in the test, call the correct function based on the param.

It does feel a bit weird to use parametrize for control flow, but I had to do a similar thing as well recently and couldn't find a better way:
https://github.com/skypilot-org/skypilot/blob/db2c05dcce1bdbe095b652c2902a1627aca435c8/tests/test_global_user_state.py#L290-L291